File: old-src/Mapping.cpp
#include "Mapping.h"

Position::Position() {}
Position::Position(int line, int column) : line{line}, column{column} {

}

Mapping::Mapping() {}
Mapping::Mapping(Position generated, Position original, int source, int name) : generated{generated},
                                                                                original{original}, source{source},
                                                                                name{name} {

}

IndexedMapping::IndexedMapping() {}
IndexedMapping::IndexedMapping(Position generated, Position original, std::string source, std::string name) : 
	generated{generated}, original{original}, source{source}, name{name} {

}

File: parcel_sourcemap/src/mapping.rs
use rkyv::{Archive, Deserialize, Serialize};

#[derive(Archive, Serialize, Deserialize, Debug, Clone, Copy, PartialEq)]
pub struct OriginalLocation {
    pub original_line: u32,
    pub original_column: u32,
    pub source: u32,
    pub name: Option<u32>,
}

impl OriginalLocation {
    pub fn new(original_line: u32, original_column: u32, source: u32, name: Option<u32>) -> Self {
        Self {
            original_line,
            original_column,
            source,
            name,
        }
    }
}

#[derive(Archive, Serialize, Deserialize, Debug, PartialEq)]
pub struct Mapping {
    pub generated_line: u32,
    pub generated_column: u32,
    pub original: Option<OriginalLocation>,
}
